modules/solr/src/main/java/org/testcontainers/containers/SolrClientUtils.java

```diff
@@ -0,0 +1,124 @@
+     * @param configurationName the name of the configuration which should be created
+     * @param solrConfig        the url under which the solrconfig.xml can be found
+     * @param solrSchema        the url under which the schema.xml can be found or null if the default schema should be used
+     */
+    public static void uploadConfiguration(int port, String configurationName, URL solrConfig, URL solrSchema) throws URISyntaxException, IOException {
+        Map<String, String> parameters = new HashMap<>();
+        parameters.put("action", "UPLOAD");
+        parameters.put("name", configurationName);
+        HttpUrl url = generateSolrURL(port, Arrays.asList("admin", "configs"), parameters);
+
+        byte[] configurationZipFile = generateConfigZipFile(solrConfig, solrSchema);
+        executePost(url, configurationZipFile);
+
+    }
+
+    /**
+     * Creates a new collection
+     *
+     * @param port              The Port on which solr is running
+     * @param collectionName    the name of the collection which should be created
+     * @param configurationName the name of the configuration which should used to create the collection
+     *                          or null if the default configuration should be used
+     */
+    public static void createCollection(int port, String collectionName, String configurationName) throws URISyntaxException, IOException {
+        Map<String, String> parameters = new HashMap<>();
+        parameters.put("action", "CREATE");
+        parameters.put("name", collectionName);
+        parameters.put("numShards", "1");
+        parameters.put("replicationFactor", "1");
+        parameters.put("wt", "json");
+        if (configurationName != null) {
+            parameters.put("collection.configName", configurationName);
+        }
+        HttpUrl url = generateSolrURL(port, Arrays.asList("admin", "collections"), parameters);
+        executePost(url, null);
+    }
+
+    private static void executePost(HttpUrl url, byte[] data) throws IOException {
+
+        RequestBody requestBody = data == null ?
+            RequestBody.create(MediaType.parse("text/plain"), "") :
+            RequestBody.create(MediaType.parse("application/octet-stream"), data);
+        ;
+
+        Request request = new Request.Builder()
+            .url(url)
+            .post(requestBody)
+            .build();
+        Response response = httpClient.newCall(request).execute();
+        if (!response.isSuccessful()) {
+            String responseBody = response.body() != null ? response.body().string() : "";
+            throw new SolrClientUtilsException(response.code(), "Unable to upload binary\n" + responseBody);
+        }
+    }
+
+    private static HttpUrl generateSolrURL(int port, List<String> pathSegments, Map<String, String> parameters) throws URISyntaxException {
+        HttpUrl.Builder builder = new HttpUrl.Builder();
+        builder.scheme("http");
+        builder.host("localhost");
```

Comment:
As above, we can't guarantee that containers are reachable on `localhost` in every environment, so we'll need to pass the hostname through along with the port.